### ISSUE TITLE
Please add table of contents

### ISSUE THREAD

**Issue by xfzka** [LukasJoswiak/etch#2]

**Comment by LukasJoswiak** [comment]
Hi, can you expand on what you mean by this?

**Comment by xfzka** [comment]
Sorry my English is bad. 
This function may call TOC.
Here is the official document https://gohugo.io/content-management/toc/

look like this
<img width="1065" alt="image" src="https://user-images.githubusercontent.com/24959175/83350482-fa57a780-a36e-11ea-9655-54eec632dba7.png">

or this
<img width="1020" alt="image" src="https://user-images.githubusercontent.com/24959175/83350565-7a7e0d00-a36f-11ea-8b9d-d79aca53f43a.png">

**Comment by LukasJoswiak** [comment]
I've attached a PR, please review it and let me know if it's what you were getting at.

**Comment by xfzka** [comment]
Thank you for you code, but i download branch of the TOC , nothing change, there is no TOC on the article page.

<img width="810" alt="image" src="https://user-images.githubusercontent.com/24959175/83626116-90860a80-a5c7-11ea-84da-45c6e00d71ec.png">

<img width="963" alt="image" src="https://user-images.githubusercontent.com/24959175/83626355-da6ef080-a5c7-11ea-81de-4a391d72921f.png">

**Comment by LukasJoswiak** [comment]
You’ll need to add `{{< toc >}}` in the Markdown where you want to insert the table of contents.

**Comment by xfzka** [comment]
Thank you, add `{{< toc >}}` to my Markdown article is work very well 😊.

Can you please make the TOC auto generate?

**Comment by LukasJoswiak** [comment]
By auto generate do you mean automatically generate for every page without using `{{< toc >}}`? Or do you mean it should auto generate titles? It should use the header elements in your article to create the table of contents.

**Comment by xfzka** [comment]
Yes, I mean **automatically generate for every page without using `{{< toc >}}`**.
If you don't like this feature,please add a config option to control it, many people like me love TOC feature.

**Comment by LukasJoswiak** [comment]
Thanks for the request. I've thought about this, but the reason I added the shortcode to generate a table of contents was to allow it to be placed anywhere on the page. The added  complexity of a global table of contents flag would create strange interactions if the page already has the table of contents shortcode, so I'm not going to add this as an option. Closing this issue and merging the associated PR.
